e2e/nomostest/sync-status-watch-controller.go

```
@@ -0,0 +1,115 @@
+
+	"kpt.dev/configsync/pkg/core/k8sobjects"
+)
+
+// StatusWatchNamespace is the namespace for the sync status watch controller.
+const StatusWatchNamespace = "sync-status-watch"
+
+// StatusWatchName is the name of the sync status watch controller.
+const StatusWatchName = "sync-status-watch"
+
+// ImagePlaceholder is the placeholder in the manifest that will be replaced with the actual image.
+const ImagePlaceholder = "SYNC_STATUS_WATCH_CONTROLLER_IMAGE"
+
+// ManifestPath is the path to the manifest file.
+const ManifestPath = "../../examples/post-sync/sync-watch-manifest.yaml"
+
+// TestImage is the image used for testing the sync status watch controller.
+const TestImage = testing.TestInfraArtifactRepositoryAddress + "/sync-status-watch-controller:v1.0.0-6ea8969a"
+
+// SetupSyncStatusWatchController sets up the sync status watch controller in the cluster.
+// It creates the necessary namespace and deploys the controller.
+func SetupSyncStatusWatchController(nt *NT) error {
+	nt.T.Logf("applying sync status watch controller manifest")
+
+	// Create namespace first to ensure it exists
+	namespace := testSyncStatusWatchNamespace()
+	if err := nt.KubeClient.Create(namespace); err != nil && !apierrors.IsAlreadyExists(err) {
+		return fmt.Errorf("creating namespace %s: %w", namespace.Name, err)
+	}
+
+	if err := execManifestCommand(nt, "apply"); err != nil {
+		nt.describeNotRunningTestPods(StatusWatchNamespace)
+		return fmt.Errorf("applying sync status watch manifest: %w", err)
+	}
+
+	return nt.Watcher.WatchForCurrentStatus(kinds.Deployment(), StatusWatchName, StatusWatchNamespace)
+}
+
+// TeardownSyncStatusWatchController tears down the sync status watch controller from the cluster.
+// It removes the namespace and associated resources.
+func TeardownSyncStatusWatchController(nt *NT) error {
+	nt.T.Log("tearing down sync status watch controller")
+
+	if err := execManifestCommand(nt, "delete"); err != nil {
+		nt.T.Logf("Error deleting resources: %v", err)
```

should probably return an error.

---

done